plugin/nginx_plus_collectd.py

```diff
@@ -673,24 +697,28 @@ class NginxStatusAgent(object):
     '''
     Helper class for interacting with a single NGINX+ instance.
     '''
-    def __init__(self, status_host=None, status_port=None, username=None, password=None):
+    def __init__(self, status_host=None, status_port=None, username=None, password=None, api_version=None, api_base_path=None):
         self.status_host = status_host or 'localhost'
         self.status_port = status_port or 8080
         self.auth_tuple = (username, password) if username or password else None
+        self.api_version = api_version if api_version is not None else DEFAULT_API_VERSION
```

Comment:
As stated earlier, I would treat unset api versions as an instruction to detect legacy or newer and set 3 as the default for newer.

Reply:
Yes, based on the latest implemented logic, unset APIVersion will lead the code to detect versioned or newer API and will set 3 as the default APIVersion for versioned API. A detailed explanation of the current logic is provided in the [above](https://github.com/signalfx/collectd-nginx-plus/pull/8#discussion_r760223262) comment.